verilog/tools/kythe/indexing_facts_tree_extractor.cc

```diff
@@ -1213,7 +1216,7 @@ void IndexingFactsTreeExtractor::ExtractTypeDeclaration(
   }
 }
 
-void IndexingFactsTreeExtractor::CopyAndDeleteLastNode(
+void IndexingFactsTreeExtractor::CopyAndDeleteLasSibling(
```

Comment:
Suggest change to a non-const reference:

```c++
IndexingFactNode& previous_node =
      facts_tree_context_.top().Children().back();
```

and if `new_node`'s anchors are empty, swap instead of copy:

```c++
new_node.Value().SwapAnchors(previous_node.Value());  // define using vector::swap()
```

```c++
new_node.AdoptSubtreesFrom(&previous_node);
```

because you're deleting `previous_node` anyhow via `pop_back()`, so move its children rather than copy, which is more efficient.